openhtf/io/http_api.py

```diff
@@ -119,14 +121,25 @@ def do_GET(self):  # pylint: disable=invalid-name
       """
       result = {'test': util.ConvertToBaseTypes(self.executor.GetState()),
                 'framework': util.ConvertToBaseTypes(self.executor)}
-      self.wfile.write(json.dumps(result))
+      with open(os.devnull, 'w') as f:
+        with contextlib.redirect_stderr(f):  # Don't spam the terminal.
+          self.send_response(200)
+          self.end_headers()
+          self.wfile.write(json.dumps(result))
 
     def do_POST(self):  # pylint: disable=invalid-name
       """Parse a prompt response and send it to the PromptManager."""
-      raw = self.rfile.read()
+      length = int(self.headers.getheader('content-length'))
+      raw = self.rfile.read(length)
       data = json.loads(raw)
       user_input.get_prompt_manager().Respond(
           uuid.UUID((data['id'])), data['response'])
+      with open(os.devnull, 'w') as f:
+        with contextlib.redirect_stderr(f):  # Don't spam the terminal.
+          self.send_response(200)
+          self.end_headers()
+          self.wfile.write('OK')
```

Comment:
might want to update `PromptManager.Respond()` to indicate if the response was actually used and propagate that info here - it would be pretty confusing if someone is at the frontend and entered a value and then that value was ignored in favor of a seemingly random one.